src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1356,7 +1357,10 @@ def create_policy_assignment(cmd, policy=None, policy_set_definition=None,
             identity = _build_identities_info(cmd, assign_identity)
         assignment.identity = identity
 
-    createdAssignment = policy_client.policy_assignments.create(scope, name or uuid.uuid4(), assignment)
+    if name is None:
+        name = (base64.urlsafe_b64encode(uuid.uuid4().bytes).decode())[:-2]
```

Comment:
>uuid4 [](start = 46, length = 5)

why not just cut the uuid?

Reply:
Because the name generated by using Python's uuid.uuid4() is 36 characters, which exceeds the limit of 24 characters required by server. So I used Base64 to compress the UUID to generate a unique random name with a length of 22.

Reply:
[:-22]?